src/libData/AccountData/EvmClient.h

```diff
@@ -23,36 +23,33 @@
 
 #include <jsonrpccpp/client.h>
 #include <jsonrpccpp/client/connectors/unixdomainsocketclient.h>
-
 #include "common/Constants.h"
+#include "common/Singleton.h"
 
 namespace evmproj {
 
 struct CallResponse;
 
 }
 
-class EvmClient {
+class EvmClient : public Singleton<EvmClient> {
  public:
-  static EvmClient& GetInstance() {
-    static EvmClient evmClient;
-    return evmClient;
-  }
+  EvmClient() = default;
+  virtual ~EvmClient() = default;
 
   void Init();
 
   bool CheckClient(uint32_t version,
                    __attribute__((unused)) bool enforce = false);
 
-  bool CallRunner(uint32_t version, const Json::Value& _json,
-                  evmproj::CallResponse& result,
-                  uint32_t counter = MAXRETRYCONN);
+  virtual bool CallRunner(uint32_t version, const Json::Value& _json,
+                          evmproj::CallResponse& result,
+                          uint32_t counter = MAXRETRYCONN);
 
- private:
-  EvmClient() {}
-  virtual ~EvmClient() {}
-  bool OpenServer(bool force = false);
+ protected:
+  virtual bool OpenServer(bool force = false);
 
```

Comment:
In EvmClient, are we supporting multiple versions now, as this was deliberatly forced to a single version with a note that we should get rid of the map later as we were minimising changes at the time. if we are not supporting multiple versions using the ScillaClient technique then it would be better to just ghet rid of the maps entirely and replace with single pointers.

Reply:
fair point, but would change this when we refactor